main.py

```diff
@@ -4,72 +4,100 @@
 # TODO: add support for uppercase, numbers and special characters
 # TODO: add option to input encryption key
 
+
 import sys
 
+char_list = [
+    "a",
+    "b",
+    "c",
+    "d",
+    "e",
+    "f",
+    "g",
+    "h",
+    "i",
+    "j",
+    "k",
+    "l",
+    "m",
+    "n",
+    "o",
+    "p",
+    "q",
+    "r",
+    "s",
+    "t",
+    "u",
+    "v",
+    "w",
+    "x",
+    "y",
+    "z",
+]
+
+full_key = [0, 1, 2]
+
+key_id = 0
+
+
+def rotate_cipher():
+    global key_id, full_key
+
+    if key_id == len(full_key) - 1:
+
+        key_id = 0
+
+    else:
+
+        key_id = key_id + 1
+
+
+def cipher(char_id):
+    global key_id, char_list, full_key
+
+    cipher_id = char_id + full_key[key_id]
+
+    if cipher_id >= len(char_list):
+        cipher_id = cipher_id - len(char_list)
+
+    otpt = char_list[cipher_id]
+
+    rotate_cipher()
+
+    return otpt
 
-def caesar():
-    alpha = [
-        "a",
-        "b",
-        "c",
-        "d",
-        "e",
-        "f",
-        "g",
-        "h",
-        "i",
-        "j",
-        "k",
-        "l",
-        "m",
-        "n",
-        "o",
-        "p",
-        "q",
-        "r",
-        "s",
-        "t",
-        "u",
-        "v",
-        "w",
-        "x",
-        "y",
-        "z",
-    ]
-
-    unenc = " ".join(sys.argv[1:])
-    unenc_cnt = 0
-    enc = ""
-    cipher = [1, 2, 3]
-    cipher_cnt = 0
-
-    # Check every character in user input...
-    for x in unenc:
-        alpha_cnt = 0
-        # ...against every character in character list...
-        for y in alpha:
-            # ...if a match, apply cipher and append to encrypted output.
-            if x == alpha[alpha_cnt]:
-                # Rotates through character list if using a character at the end of the list.
-                if alpha_cnt + cipher[cipher_cnt] >= len(alpha):
-                    alpha_cnt = alpha_cnt - len(alpha)
-
-                enc = enc + alpha[alpha_cnt + cipher[cipher_cnt]]
-                # Rotate through cipher.
-                if cipher_cnt == len(cipher) - 1:
-                    cipher_cnt = 0
-                else:
-                    cipher_cnt = cipher_cnt + 1
-
-            alpha_cnt = alpha_cnt + 1
-        unenc_cnt = unenc_cnt + 1
-
-    print("Unencrypted:", unenc, "\nEncrypted:", enc)
+
+def get_char_id(x):
+    global char_list
+
+    char_id = 0
+
+    for y in char_list:
+
+        if x == y:
+            return char_id
```

Comment:
To improve readability, I've renamed the X and Y

```suggestion
def get_char_id(given_char):
    global char_list

    char_id = 0

    for eval_char in char_list:

        if given_char == eval_char:
            return char_id
```